modules/touch/src/win32_touch.cpp

```diff
@@ -155,28 +213,85 @@ Win32TouchHook::Win32TouchHook(void* nativeWindow)
 
     if (!gStarted) {
         gStarted = true;
+        gStartTime = std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::high_resolution_clock::now().time_since_epoch()
+        );
+#ifdef ENABLE_TUIOMESSAGES
         gTuioServer = new TUIO::TuioServer("localhost", 3333);
         TUIO::TuioTime::initSession();
+#endif
         gTouchHook = SetWindowsHookExW(
             WH_GETMESSAGE,
             HookCallback,
             GetModuleHandleW(NULL),
             GetCurrentThreadId()
         );
+
+        // In theory, if our UI is pumped from a different thread, we can
+        // handle Low-level mouse events in that thread as well.
+        // this might help reduce mouse lag while running OpenSpace?
+        // gMouseHookThread = new std::thread([](){
+        //     gMouseHook = SetWindowsHookExW(
+        //         WH_MOUSE_LL,
+        //         LowLevelMouseProc,
+        //         GetModuleHandleW(NULL),
+        //         0 //<- Global thread id (low-level mouse is global only)
+        //     );
+        //     if(!gMouseHook){
+        //         LINFO("Could not setup mousehook!");
+        //     }
+
+        //     MSG msg;
+        //     while (GetMessage(&msg, NULL, 0, 0))
+        //     {
+        //         DispatchMessage(&msg);
+        //     }
+        // });
+
         if (!gTouchHook) {
             LINFO(fmt::format("Failed to setup WindowsHook for touch input redirection"));
+#ifdef ENABLE_TUIOMESSAGES
             delete gTuioServer;
+#endif
             gStarted = false;
         }
     }
 }
 
+
 Win32TouchHook::~Win32TouchHook() {
     if (gStarted) {
         UnhookWindowsHookEx(gTouchHook);
+        UnhookWindowsHookEx(gMouseHook);
+#ifdef ENABLE_TUIOMESSAGES
         delete gTuioServer;
+#endif
     }
 }
 
+// Low-level mouse hook is "needed" if we want to stop mousecursor from moving
+// when we get a touch-input on our window A negative effect is that this
+// function is for global threads, meaning our application will cause Windows to
+// stall the mouse cursor when this function can't be scheduled. This is not yet
+// fail-proof...might be a race-condition on message pumping?
+// - Seems to move the cursor when we get two fingers as input..
+// - If we ourselves would pump windows for events, we can handle this in the
+// pump-loop
+LRESULT CALLBACK LowLevelMouseProc(int nCode, WPARAM wParam, LPARAM lParam) {
+    if (nCode < 0) {
+        // do not process message
+        return CallNextHookEx(0, nCode, wParam, lParam);
+    }
+    LPMSLLHOOKSTRUCT msg = reinterpret_cast<LPMSLLHOOKSTRUCT>(lParam);
+    // block injected events (in most cases generated by touches)
+    if (msg->flags & LLMHF_INJECTED || msg->dwExtraInfo == 0xFF515700) {
```

Comment:
Where does the constant come from?  Better to define it with a proper name as `constexpr const` before?